hub/core/dataset/dataset.py

```diff
@@ -314,8 +314,11 @@ def create_tensor(
             version_state=self.version_state,
             **meta_kwargs,
         )
-        self.version_state["meta"].tensors.append(name)
-        ffw_dataset_meta(self.version_state["meta"])
+        meta = self.version_state["meta"]
```

Comment:
Are you sure it have to move here ? Seems previously it was part of `version_control`

Reply:
Yeah, the issue was that at times on creating new tensors the Dataset meta wasn't in dirty keys and so it wasn't getting updated in the storage (again weirdness caused due to Cachable). Now everytime we're creating a tensor, we're adding the Dataset meta back to the dirty keys in case it wasn't there. This will ensure consistency.